File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/BreakingNews.java
package com.xebia.functional.xef.java.auto;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;

public class BreakingNews {

    public String summary;

    static DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/M/yyyy");
    static LocalDateTime now = LocalDateTime.now();

    @Override
    public String toString() {
        return "BreakingNews{" +
              "summary='" + summary + '\'' +
              '}';
    }

    private static CompletableFuture<Void> writeParagraph(AIScope scope) {
        var currentDate = dtf.format(now);

        return scope.prompt("write a paragraph of about 300 words about: " + currentDate + " Covid News", BreakingNews.class)
              .thenAccept(breakingNews -> System.out.println(currentDate + " Covid news summary:\n" + breakingNews));
    }

    public static void main(String[] args) throws ExecutionException, InterruptedException {
        try (AIScope scope = new AIScope()) {
            var currentDate = dtf.format(now);
            scope.contextScope(scope.search(currentDate + " Covid News"), BreakingNews::writeParagraph).get();
        }
    }
}

File: example/src/main/kotlin/com/xebia/functional/xef/auto/BreakingNews.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.agents.search
import kotlinx.serialization.Serializable
import java.text.SimpleDateFormat
import java.util.Date

@Serializable
data class BreakingNewsAboutCovid(
  val summary: String
)

suspend fun main() {
  ai {
    val sdf = SimpleDateFormat("dd/M/yyyy")
    val currentDate = sdf.format(Date())
    context(search("$currentDate Covid News")) {
      val news: BreakingNewsAboutCovid =
        prompt("write a paragraph of about 300 words about: $currentDate Covid News")
      println(news)
    }
  }.getOrElse { println(it) }
}
